### ISSUE TITLE
[$1000] Jump to last unread chat when opening a report

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#35011]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.28.0
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @MonilBhavsar 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1705558628672569

## Action Performed:
1. A is participant of a chat or room
2. As another user send some good number of messages to chat report, so new messages can be scrolled up and down
3. As user A, open the chat report

## Expected Result:
The chat report view opens from where the messages are unread and user can scroll down

## Actual Result:
The chat report view is scrolled down to the last message and user needs to scroll up to find last read message

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [x] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

https://github.com/Expensify/App/assets/93399543/55def247-9ffe-439d-8f24-12aa35aee272

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01fc144b2f2f5df388</li>
        <li>Upwork Job ID: 1749920553887232000</li>
        <li>Last Price Increase: 2024-06-11</li>
<li>Automatic offers: </li>
<ul>
<li>ikevin127 | Contributor | 0</li>
<li>ishpaul777 | Contributor | 102670752</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @chrispader</details>

**Comment by jeremy-croff** [comment]
This PR https://github.com/Expensify/App/pull/28793 specifically makes it scrolled to the bottom, seems like NAB

**Comment by roryabraham** [comment]
HOLD for comment linking: https://github.com/Expensify/App/pull/30269

**Comment by trjExpensify** [comment]
No change, still on hold Melv!

**Comment by trjExpensify** [comment]
Samesies, Melv!

**Comment by trjExpensify** [comment]
Still on hold, comment linking got held for the deploy freeze last week.

**Comment by trjExpensify** [comment]
PR still not merged.

**Comment by trjExpensify** [comment]
Comment linking PR still in the works.

**Comment by trjExpensify** [comment]
Bumped!

**Comment by trjExpensify** [comment]
Looking for proposals. @jeremy-croff not sure if you're interested?

**Comment by trjExpensify** [comment]
Still looking for proposals!

**Comment by trjExpensify** [comment]
Bumped the price up to $750

**Comment by ikevin127** [comment]
# Proposal

### Please re-state the problem that we are trying to solve in this issue

When opening a chat with lots of unread messages, the user needs to scroll up to find last read message.

### What is the root cause of that problem?

`New Feature` - The logic is missing / non-existent.

### What changes do you think we should make in order to solve the problem?

We need a new `useEffect` within the `ReportActionsList` component which will `scrollToFirstUnreadMessage` when we open the report.

To do this we will need to modify our existing `reportScrollManager.scrollToIndex` function to optionally take in the [`viewPosition`](https://reactnative.dev/docs/flatlist#scrolltoindex) argument because we're inverting the flatlist which means that we would need to pass `viewPosition: 1` to the FlatList's `scrollToIndex` function such that we are scrolling to the top of the new message (unread marker).

<details>
<summary>New useEffect code</summary>

```ts
    useEffect(() => {
        // Logic to scroll to the first unread message, if there is one and it's not in view.
        const scrollToFirstUnreadMessage = () => {
            if (!currentUnreadMarker) {
                return;
            }

            const unreadMessageIndex = sortedVisibleReportActions.findIndex((action) => action.reportActionID === currentUnreadMarker);

            if (unreadMessageIndex !== -1) {
                // We're passing viewPosition: 1 since we're using an InvertedFlatList.
                reportScrollManager?.scrollToIndex(unreadMessageIndex, false, 1);
            }
        };

        // Call the scroll function after a small delay to ensure all items 
        // have been measured and the list is ready to be scrolled.
        InteractionManager.runAfterInteractions(scrollToFirstUnreadMessage);
        // We only want to run this effect once, when we're navigating to a report with unread messages.
        // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [sortedVisibleReportActions]);
```

</details>

Besides the above new `useEffect` code introduced as part of this new feature, here's a test branch [test/35011](https://github.com/ikevin127/Expensify/tree/test/35011) and the [diff](https://github.com/Expensify/App/compare/main...ikevin127:test/35011) in order to review all the changes introduced as part of this feature, including the `scrollToIndex` related changes for both web / native platforms.

**Additionally:** (did not add this to the current solution) we could also introduce the [`viewOffset`](https://reactnative.dev/docs/flatlist#scrolltoindex) argument to the `scrollToIndex` function if we want some additional offset such that the new unread message marker would have better visibility if in the below picture example it's not visible enough (current solution, also see videos):

<img width="380" alt="Screenshot 2024-04-27 at 18 35 53" src="https://github.com/Expensify/App/assets/56457735/cde468cf-448e-404b-b98c-3c1f0ce09091">

### Video results

<details>
<summary>MacOS: Chrome / Safari</summary>


https://github.com/Expensify/App/assets/56457735/c37e15f1-17c4-4bc7-b357-c94e01cbcc2e



</details>

<details>
<summary>iOS: Native</summary>


https://github.com/Expensify/App/assets/56457735/e4282338-310e-44d3-97b2-447ebf7d2fb6



</details>

<details>
<summary>Android: mWeb</summary>

https://github.com/Expensify/App/assets/56457735/9a00be14-b09d-40d9-bae3-4171605fe83a




</details>

**Comment by trjExpensify** [comment]
@mollfpr can you review this proposal please? CC: @roryabraham feels like one you'd be interested in as well. :)

**Comment by mollfpr** [comment]
Sorry for the delay 🙏

The @ikevin127 proposal looks good to me!

🎀 👀 🎀 C+ reviewed!

**Comment by ikevin127** [comment]
PR https://github.com/Expensify/App/pull/41448 ready for review! 🚀

**Comment by ikevin127** [comment]
^
Discussing this on C+ [Slack 🧵](https://expensify.slack.com/archives/C02NK2DQWUX/p1715609260869129).
**Note:** So far looks like false alarm -> staging server (BE) issues and not a regression of this PR / FE issue.

**Comment by ikevin127** [comment]
Based on [Slack 🧵](https://expensify.slack.com/archives/C01GTK53T8Q/p1715689870057479) we decided to revert since it's not a lot of code and we had at least one (confirmed) deploy-blocking regression @ https://github.com/Expensify/App/issues/42100, possible another one @ https://github.com/Expensify/App/issues/42148 (not confirmed yet).

**Comment by ikevin127** [comment]
This should not be on HOLD for payment as the PR was reverted:
- https://github.com/Expensify/App/pull/42167

meaning no payment will be issued here yet.

I will post a new proposal today with a solid re-work of the solution, hoping we can start from scratch here and build something solid this time. Will have to test it better this time, carefully considering issues that were reported after first PR was merged like:
- https://github.com/Expensify/App/issues/42100 (confirmed, see [#42100 (comment)](https://github.com/Expensify/App/issues/42100#issuecomment-2110896094))
- https://github.com/Expensify/App/issues/42148 (confirmed, see [#42148 (comment)](https://github.com/Expensify/App/issues/42148#issuecomment-2113133353))
- making sure any other report scrolling functionality will not be affected by the new solution

cc @trjExpensify @yuwenmemon @mollfpr

**Comment by ikevin127** [comment]
# Proposal

### Please re-state the problem that we are trying to solve in this issue

When opening a chat with lots of unread messages, the user needs to scroll up to find last read message.

### What is the root cause of that problem?

`New Feature` - The logic is missing / non-existent.

### What changes do you think we should make in order to solve the problem?

**First we need to understand how our `InvertedFlatList` works when a report with multiple actions is open:**
1. Our `ReportActionsList` -> InvertedFlatList is using the `initialNumToRender` prop where we calculate the ideal number of report actions to render in the first paint (first component mount).
2. ^ This means that the first paint of the open report renders anywhere between 10-20 report actions, depending on device height, see code [here](https://github.com/Expensify/App/blob/82bacc94d7601d3e10e6c7586e0cb9d04356b072/src/pages/home/report/ReportActionsList.tsx#L417-L429), and any subsequent report actions are rendered in a waterfall way (sync, one after the other).
3. ^ This means that right when we mount the `ReportActionsList`, even if we identify the report action with the unread marker, if this action's index > initialNumToRender, this means that the scrolling won't work since the FlatList did not render this report action yet.

**In order to deal with the complexity of this issue I came up with the following solution:**
1. First, we need a way to detect when the FlatList report action at a specific index was rendered so we can scroll to it if the unread marker was rendered. To do this I passed down a custom function the `ReportActionsListItemRenderer` onLayout prop, adding the `onLayout` prop to: ReportActionsListItemRenderer -> ReportActionItemParentAction & ReportActionItem.

By doing this we can pass the following custom function to the report action item:
```ts
    function onItemLayout(index: number) {
        console.log('Rendered item at index ->', index + 1);
        DeviceEventEmitter.emit('renderedItemIndex', index);
    }
```
which means that when a certain report action item was indeed rendered within the FlatList, we emit the index of the rendered report action so we are aware of it. This is how the console log looks when opening a report:

<details>
<summary>MacOS: Chrome</summary>

https://github.com/Expensify/App/assets/56457735/e5aa56f0-ab88-4ab0-aef0-30ac00d8dfb0

</details>

2. The second part is `DeviceEventEmitter`, I went with this method since in the next part we will use `DeviceEventEmitter.addListener('renderedItemIndex', scrollToUnreadMessageIndex)` in order to scroll to the report action unread marker ONLY once the report action was actually rendered within the FlatList, and by using the listener, we don't need to add any rendered items variable to the useEffect dependency array.
3. We need a new `useEffect` within the `ReportActionsList` component which will `scrollToUnreadMessageIndex` when we open the report, using the DeviceEventEmitter listener.

<details>
<summary>New useEffect Code</summary>

```ts
    useEffect(() => {
        function scrollToUnreadMessageIndex(index: number) {
            // If there is no unread marker, we don't need to scroll.
            if (!currentUnreadMarker) {
                return;
            }

            const unreadMessageIndex = sortedVisibleReportActions.findIndex((action) => action.reportActionID === currentUnreadMarker);
            // Scroll to unread marker if:
            // 1. We found the unread message index (index !== -1).
            // 2. The unread message index was rendered (index === unreadMessageIndex).
            if (unreadMessageIndex !== -1 && index === unreadMessageIndex) {
                // Scroll after interactions to ensure the list is ready to be scrolled.
                InteractionManager.runAfterInteractions(() => {
                    // We're passing viewPosition: 1 to scroll to the top (inverted FlatList)
                    // of the unread message marker.
                    reportScrollManager?.scrollToIndex(unreadMessageIndex, false, 1);
                });
            }
        }

        const subscription = DeviceEventEmitter.addListener('renderedItemIndex', scrollToUnreadMessageIndex);
        return () => {
            subscription.remove();
        };
        // When we're navigating to a report with unread messages, we only want to run this effect once,
        // after the unread marker is set and the item at the unread marker index is rendered.
        // Dependency array reasoning:
        // [+] currentUnreadMarker - We need to keep this updated, otherwise the effect will scroll to
        // previously unread marker since this state is initialized from cacheUnreadMarkers.
        // [-] reportScrollManager - Doesn't update.
        // [-] sortedVisibleReportActions - Don't re-run this effect if the list of visible report actions updates.
    // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [currentUnreadMarker]);
```

</details>

4. Finally, we need to update our existing `reportScrollManager.scrollToIndex` function / types to optionally take in the [`viewPosition`](https://reactnative.dev/docs/flatlist#scrolltoindex) argument because we're inverting the flatlist which means that we would need to pass `viewPosition: 1` to the FlatList's `scrollToIndex` function such that in the above `useEffect` we are scrolling to the top of the new message (unread marker).

### Video results

<details>
<summary>MacOS: Chrome</summary>

https://github.com/Expensify/App/assets/56457735/591920bd-7025-4e46-88f7-564227525af4

</details>

**Comment by ikevin127** [comment]
New PR open with solution re-work:
- https://github.com/Expensify/App/pull/42217

@mollfpr Checkout the new [proposal](https://github.com/Expensify/App/issues/35011#issuecomment-2113414454) and let me know if you have any code-related questions during PR review that are not answered by the proposal. If there are any other issues related to the PR, we can discuss them there.

**Comment by trjExpensify** [comment]
Removed the payment hold as per [here](https://github.com/Expensify/App/issues/35011#issuecomment-2113140457).

**Comment by trjExpensify** [comment]
How's the PR review coming along, @mollfpr?

**Comment by ikevin127** [comment]
@trjExpensify Can we re-assign C+ here for review ?

**Recap:**
- first PR was reverted
- new (reworked) PR was open **23 days ago**
- reviewer commented "I'll review today" **10 days sgo**

To me it seems like there's no capacity on reviewer's side to move this along. What do you think ?

**Note:** I think re-assignment will be easy here considering the bounty and the fact that the first PR was reverted means that neither me or the current reviewer are owed anything so far.

**Comment by mollfpr** [comment]
Sorry for the inconvenience. I couldn't get the review done because I need more time on testing and currently more PR coming to get reviewed, so I can't prioritize this issue last few days.

I'll unassign myself and I apologize for the delay.

**Comment by mollfpr** [comment]
@ishpaul777 will be taking over this issue.

**Comment by ikevin127** [comment]
When any of you get a change, could you please assign **[ishpaul777](https://github.com/ishpaul777)** to the **issue** and **PR** https://github.com/Expensify/App/pull/42217 as **reviewer** ? Thank you!

**Status update:** 
Reviewing was picked-up by **ishpaul777** and we're looking at having the checklist completed this weekend, after thorough testing.


cc @trjExpensify @yuwenmemon

**Comment by trjExpensify** [comment]
Ishpaul is assigned to this issue now!

**Comment by ikevin127** [comment]
I closed the new PR since after further testing / review I came to the conclusion that my proposal is far from addressing this `NewFeature`, see more details in https://github.com/Expensify/App/pull/42217#issuecomment-2159592684. This is more complex than I thought initially, way over my head thb. Good luck to the next contributor and feel free to use any of my previous proposal / code!

@trjExpensify I guess this can go back to looking for proposals and maybe even a bounty increase since this is not an easy feat.

**Comment by trjExpensify** [comment]
>@trjExpensify I guess this can go back to looking for proposals and maybe even a bounty increase since this is not an easy feat.

Thanks for the heads up, @ikevin127. I've gone ahead and opened it back up.

**Comment by brunovjk** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
The chat report view is scrolled down to the last message and user needs to scroll up to find last read message

### What is the root cause of that problem?
This is part of the "Comment linking". Where do we configure it to scroll to the bottom if there is no linked report:
https://github.com/Expensify/App/blob/7c76431d206fce37da08b29cbbd7db802cbb55b7/src/pages/home/report/ReportActionsList.tsx#L332-L340

### What changes do you think we should make in order to solve the problem?
We need to update to `scrollToIndex` of  the unread message, if any, otherwise `scrollToBottom`, when enter a chat report.

**Note**: Although simple, we need a few additional weeks just to test, make sure we don't ruin anything about the "Comment Linking" system or the app as a whole.

### What alternative solutions did you explore? (Optional)

**Comment by ikevin127** [comment]
Not sure, it's complicated and most likely will require BE changes for the `OpenReport` call given our current logic.
What I'd do here is I'd bring this up w/ the community on Slack and get some feedback from the Expensify team regarding which way we should go about this since it's a `NewFeature`.

##

> i think a potential solution could be implemented if the openReport Call returns the unseen messages from top means there 160 message will always have the first unread message then we could modify the logic to retrieve the newer messages as user scroll down to newer messages, wdyt @ikevin127

I think this **could** work given the way we use the FlatList currently and the fact that when `OpenReport` is called we only get ~168 report actions which are rendered in a waterfall way as user scrolls up and we have `GetOlderActions` as we scroll / reach close to the first ~168 report actions batch -> which will fetch the next ~168 report actions.

**Comment by brunovjk** [comment]
Thank you for the clarification, sorry about that. I will keep investigating.

**Comment by ikevin127** [comment]
No, I did not. Feel free do to that if it can't be sorted out here and you think a broader discussion would help bring this closer to completion.

**Comment by trjExpensify** [comment]
Done!

**Comment by trjExpensify** [comment]
Taking the issue off hold.

**Comment by roryabraham** [comment]
My suggestion for next steps:

- Right now, when OpenReport is called w/o any specific reportActionID as an anchor, we essentially return the 50 newest visible actions on the report. We'll want to update that such that it looks at the `clientLastReadTime` (already passed by App) and use that to determine the anchor action. It will then return 25 messages before and 25 messages after that anchor, and the client will render those
- Then we already have some logic to fetch newer actions as the user scrolls down

I'll need to test carefully to make sure that making the back-end changes won't break NewDot without any changes. If it won't then we're good here. If it will, some additional steps and a more careful rollout might be needed.

**Comment by trjExpensify** [comment]
What's the latest here?

**Comment by roryabraham** [comment]
Was looking into this, chats are fetched in OpenReport from Web-E [here](https://github.com/Expensify/Web-Expensify/blob/4c98a24b2a444a56719c7b4d53111183604e9412/lib/ReportAPI.php#L5616-L5617)

**Comment by roryabraham** [comment]
ok, I tried just making the change to pass the `clientLastReadTime` as the anchor action, but it didn't work. Seems like NewDot changes will be neded

**Comment by roryabraham** [comment]
So one thing I'm seeing is that [this code](https://github.com/Expensify/App/blob/74375cfe6f5b88963676c8dc7c4e6b1b2502fe19/src/libs/Middleware/Pagination.ts#L109-L111) is problematic. Because we don't have access to the `lastReadReportActionID`. (only the `lastReadTime`, which will tend to be more reliable, because what if the `lastReadReportActionID` were deleted while you were offline or something?).

Basically, that code was built with the (then-correct) assumption that initially opening a report without a cursor action would mean that you'll get the latest actions. However, this issue seeks to change that. To correctly track pagination with this change, we need to return some additional pagination metadata. Fortunately, I've already got an issue for that [here](https://github.com/Expensify/App/issues/41153). So I'm going to put this on HOLD for that

**Comment by roryabraham** [comment]
still working on https://github.com/Expensify/App/issues/41153

**Comment by trjExpensify** [comment]
Yep, agreed!

**Comment by roryabraham** [comment]
All the PRs for the back-end of https://github.com/Expensify/App/issues/41153 have been merged and should go to prod on Wednesday 2024-09-18

**Comment by suneox** [comment]
I'll update my [prototype branch](https://github.com/suneox/ExpensifyApp/tree/35011-implement-report-last-read) reflect to the latest main

**Comment by roryabraham** [comment]
Going to reassign this to another Expensify engineer for the front-end parts.

**Comment by roryabraham** [comment]
Thanks @rlinoz

**Comment by rlinoz** [comment]
@janicduplessis can I assign you to this one?

**Comment by janicduplessis** [comment]
@rlinoz Yes!

**Comment by janicduplessis** [comment]
@rlinoz Is there a way I can test what you mentioned [here](https://github.com/Expensify/App/pull/49958/files#r1792363760)? I can look at the issues you encountered and then it sounds like this would be solved.

**Comment by rlinoz** [comment]
Just sent you a DM with how to access that

**Comment by rlinoz** [comment]
We have draft PRs for backend and frontend

**Comment by janicduplessis** [comment]
The PR is mostly working already, but there is an edge case that will require some refactoring of ReportActionsView / ReportActionsList so we can reuse the logic to start the list at a specific reportActionID. I will work on this in a separate PR to avoid making the current one giant.

**Comment by janicduplessis** [comment]
Have a new PR up to refactor how we handle opening the list at a specific index. Before it was mixed with the comment linking code and now it is extracted out to BaseInvertedFlatList so it will make it easy to use for opening the report at last unread message too.

https://github.com/Expensify/App/pull/52149, currently in draft, will test more and complete checklist later today.

**Comment by trjExpensify** [comment]
@janicduplessis is the PR ready for a re-review from @ishpaul777?

**Comment by janicduplessis** [comment]
Yes, I have fixed the issue he found

**Comment by trjExpensify** [comment]
What's the latest here?

**Comment by rlinoz** [comment]
https://github.com/Expensify/App/pull/52149#issuecomment-2502684929

@janicduplessis is finding time to fix a bug @ishpaul777 found in the first PR

**Comment by janicduplessis** [comment]
https://github.com/Expensify/App/pull/52149 is ready. After that it will make implementing the functionality simple

**Comment by johncschuster** [comment]
@janicduplessis how's this one going?

**Comment by rlinoz** [comment]
First PR is in the last steps I think, just some lint fix missing and I will review it again

**Comment by janicduplessis** [comment]
Working on a fix for the regression: https://github.com/Expensify/App/pull/55582

Also working on https://github.com/Expensify/App/pull/51366/ to finalize this feature, it seems to work well, but have some failing tests to fix still.

**Comment by rlinoz** [comment]
We need one more PR here, so no payment just yet

**Comment by janicduplessis** [comment]
Had to deal with a little regression from https://github.com/Expensify/App/pull/52149, last PR should be done Friday or Monday.

**Comment by trjExpensify** [comment]
How we looking here?

**Comment by rlinoz** [comment]
FE Draft PR is here https://github.com/Expensify/App/pull/51366

But we will need a backend PR to continue, I will work on that today

**Comment by rlinoz** [comment]
PRs in review

**Comment by rlinoz** [comment]
Last PR Web-E PR was merged today, so we probably can continue with the PR tomorrow @janicduplessis

**Comment by rlinoz** [comment]
@janicduplessis PR is in staging

**Comment by janicduplessis** [comment]
Working on last adjustments as per https://github.com/Expensify/App/pull/51366#issuecomment-2619681836, then will need to do a good round of testing. If all goes well I am expecting it to be ready by start of next week.

**Comment by trjExpensify** [comment]
Great stuff!

**Comment by rlinoz** [comment]
Hey @janicduplessis any updates here?

**Comment by janicduplessis** [comment]
Sorry about the delay, I didn't end up having much time to work on this this week. It should be ready by early next week :)

**Comment by janicduplessis** [comment]
Still working on this, I hit a little issue where for some reason the first unread action is now wrong. It seems to be a regression since it happened after merging main. Trying to pinpoint what caused this.

**Comment by janicduplessis** [comment]
Good news, after updating to latest main the regression is gone. PR https://github.com/Expensify/App/pull/51366 is now ready for review and further testing!

**Comment by rlinoz** [comment]
@chrispader took over while @janicduplessis is OOO, we are still ironing out some bugs.

**Comment by trjExpensify** [comment]
What's the latest here? We've had this issue/PR open a while at this point.

**Comment by rlinoz** [comment]
@chrispader can you give us an update here?

**Comment by chrispader** [comment]
I'm working on it right now. Basically the/a problem right now is, that when we have an unread report, then go offline, and then open the report, the messages and report will still be marked as read. This messes up the logic where we jump to the last unread message, once the user goes online and the rest of the messages loads.

I think i've found a potential solution, but still have to work on some edge-cases

**Comment by rlinoz** [comment]
PR is ready for a new review, we are looking whether there is a new bug or not.

**Comment by trjExpensify** [comment]
Looks like the PR is still in review. Also, not sure if this has been considered already, but you might want to test multi-expense reports with the `tableReportView` beta as that's super close to being released. CC: @Expensify/design @mountiny @luacmartins

**Comment by shawnborton** [comment]
Good shout Tom!

**Comment by rlinoz** [comment]
@trjExpensify is the expected behavior that it always ope at the first message?

**Comment by rlinoz** [comment]
hahaha fair

in the `tableReportView`, given an expense report with multiple expenses and 100 comments, should it always open at the top showing the expenses of the report or at the last unread message?

**Comment by rlinoz** [comment]
@chrispader given the comment above, from what I was testing we are always opening at the top and load all the messages in between the unread and the oldest one. Can you also take a look at that, please?

**Comment by rlinoz** [comment]
Quick update here, we are currently facing a bug that occurs only on Firefox, where the chat stops scrolling. @chrispader is looking into it.

**Comment by chrispader** [comment]
> Quick update here, we are currently facing a bug that occurs only on Firefox, where the chat stops scrolling. [@chrispader](https://github.com/chrispader) is looking into it.

@rlinoz This last Firefox issue should be fixed now too hopefully!

**Comment by trjExpensify** [comment]
@rlinoz @chrispader what's the latest here?

**Comment by chrispader** [comment]
> [@rlinoz](https://github.com/rlinoz) [@chrispader](https://github.com/chrispader) what's the latest here?

@trjExpensify we're sorting out the last issues with the PR. @ishpaul777 pointed out two issues that i couldn't reproduce, so we'll have to check whether we have to fix anything or whether we can merge this already!

**Comment by rlinoz** [comment]
Both backend PRs are merged, should get deployed today @chrispader

**Comment by rlinoz** [comment]
All the PRs are in staging, so it should be good to test.

**Comment by rlinoz** [comment]
Found another problem in the backend, new PR got merged here https://github.com/Expensify/Auth/pull/16226 will be deployed on Monday so we can continue.

**Comment by johncschuster** [comment]
How's this one coming, @rlinoz?

**Comment by rlinoz** [comment]
Still trying to fix one of the encoutered issues in a new PR to make it easier to review/merge

**Comment by johncschuster** [comment]
@chrispader how's this going?

**Comment by chrispader** [comment]
All issues from a QA run are fixed, last things i have to work on is fix the failing tests in `UnreadIndicatorsTest`. Other than that, the PR is ready to merge!

**Comment by rlinoz** [comment]
We got some new reviews in the PR, @chrispader can you take a look please?

**Comment by ishpaul777** [comment]
sure @trjExpensify i will start reviewing finishing this off 🙇

**Comment by chrispader** [comment]
@rlinoz i'll try to get this fixed tomorrow!

**Comment by chrispader** [comment]
@rlinoz Continued working on fixing edge-cases and verifying the current implementation in https://github.com/Expensify/App/pull/51366

**Comment by chrispader** [comment]
Gave an update on this issue in [this comment](https://github.com/Expensify/App/pull/51366#issuecomment-4227179218). Basically i only need to make sure whether we have conflicts with @aimane-chnaif's changes above and whether all mentioned edge-cases mentioned in this issue are covered.

**Comment by rlinoz** [comment]
But I mean, if their issue is scrolling to last unread, then this is a dupe of that no?

**Comment by aimane-chnaif** [comment]
The action badge deep link (Case 1) will be based on implementation of last unread message deep link (Case 2) which is this issue. Not dupe.

<img width="821" height="192" alt="Image" src="https://github.com/user-attachments/assets/2839e855-7629-4909-b2b4-fbb1c2347628" />

**Comment by rlinoz** [comment]
Cool, let's finish this one then, thanks! 🚀

**Comment by chrispader** [comment]
> Please make sure that [this bug](https://github.com/Expensify/App/issues/85744) doesn't happen when deep link to last unread message.
> 
> Repro step:
> 
> Preconditions:
> 
> * Have a workspace with Expensify cards enabled.
> * Have a draft expense report in the Expensify cards workspace.
> 
> 1. [web] Open the Expensify app
> 2. [web] Issue an Expensify virtual card to yourself
> 3. [web] Go to workspace chat
> 4. [web] There's a system message indicating that the Expensify card was assigned to you.
> 5. [web] Mark report preview as unread
> 6. [native] Login with same account and go to the same workspace chat
> 7. [native] **Verify that there's a system message indicating that the Expensify card was assigned to you.**

@aimane-chnaif i've verified that this bug is not reproducible with my changes in https://github.com/Expensify/App/pull/51366. Here you can see the system message:

<img width="1206" height="2622" alt="Image" src="https://github.com/user-attachments/assets/8e72c34e-d9ef-4fbe-add7-4440d7f4bb27" />

**Comment by chrispader** [comment]
@aimane-chnaif is there anything else that i need to be aware of, so this these changes are compatible with the new LHN action badges feature?

**Comment by aimane-chnaif** [comment]
> [@aimane-chnaif](https://github.com/aimane-chnaif) is there anything else that i need to be aware of, so this these changes are compatible with the new LHN action badges feature?

Does your PR just scroll to the last unread chat or also deep link?

"Deep link" here means that report action is highlighted:

<img width="1124" height="743" alt="Image" src="https://github.com/user-attachments/assets/cc8896f3-4994-4d2c-96a1-1aedbb7fe438" />

**Comment by aimane-chnaif** [comment]
We're going to deep link from LHN action badge.

https://github.com/Expensify/App/pull/87166/changes#diff-c167a2c2d0101ddeac25514c74f96571ea498e12795dea5e0c5b65910a4664ecR74

**Comment by chrispader** [comment]
> Does your PR just scroll to the last unread chat or also deep link?
>
> "Deep link" here means that report action is highlighted:

@aimane-chnaif my PR does not change the behavior of deep-linking itself, which is already working on `main` at the moment, it only changes how the chat starts rendering when a specific report action is provided. In both cases - unread and deep-linked message - the report will start rendering the list from that message.

Any of your changes needed for deep-linking from the LHN action badges should therefore not conflict with my changes. As long as a `:reportActionID` route param is provided when navigating to the report, the app will deep-link to that message.

**Comment by trjExpensify** [comment]
Not entirely sure why I was ⭐'d on this issue. Putting it back on Chris who's leading this issue.

**Comment by mitarachim** [comment]
Deploy Blocker https://github.com/Expensify/App/issues/90032 was identified to be related to the PR https://github.com/Expensify/App/pull/51366 that attempted to fix this issue.

**Comment by rlinoz** [comment]
Btw this should not have been closed?

**Comment by lanitochka17** [comment]
Deploy Blocker #https://github.com/Expensify/App/issues/90052 was identified to be related to the PR #https://github.com/Expensify/App/pull/51366 that attempted to fix this issue.

**Comment by jponikarchuk** [comment]
Deploy Blocker #90064 was identified to be related to the PR #51366 that attempted to fix this issue.

**Comment by chrispader** [comment]
Follow-up PR with a fix for the above deploy blockers is ready!

- https://github.com/Expensify/App/pull/90072

**Comment by rlinoz** [comment]
We have somew follow up issues, but we are finally done with this one!

https://github.com/Expensify/App/issues/89676
https://github.com/Expensify/App/issues/89677

**Comment by chrispader** [comment]
Awesome, i'm super stocked! 🎉

**Comment by ishpaul777** [comment]
Hey! just noticing that i might be due payment as seperate for this now that this is done.

See detailed issue summary see this [comment](https://github.com/Expensify/App/issues/35011#issuecomment-3490489365), with reviews done both before and after I joined the Backend C+ team. None of the review work on this issue has been logged to my hourly tracking, since I was already assigned here before joining the team and wanted to keep it separated.

**PRs I reviewed under this issue:**

- [#42217](https://github.com/Expensify/App/pull/42217) (+ proposal reviews after it was closed)
- [#52149](https://github.com/Expensify/App/pull/52149)
- [#51366](https://github.com/Expensify/App/pull/51366)
- [#76101](https://github.com/Expensify/App/pull/76101)
- [#76104](https://github.com/Expensify/App/pull/76104)
- [#83932](https://github.com/Expensify/App/pull/83932)

Given the scope and timeline, I'd propose a lump sum of *$1,500*.

cc @rlinoz @trjExpensify @mallenexpensify

**Comment by mallenexpensify** [comment]
@ishpaul777 can you review the above list? You were [onboarded to the BEC team around Nov 2nd ](https://github.com/Expensify/Expensify/issues/561934) and https://github.com/Expensify/App/pull/83932#issue-4013129736 was created on March 2nd.

**Comment by ishpaul777** [comment]
> None of the review work on this issue has been logged to my hourly tracking

hey as i said i never counted time spent on PRs related this issue to hourly tracking

**Comment by mallenexpensify** [comment]
Discussing internally..

**Comment by ishpaul777** [comment]
@mallenexpensify any updates for me?

**Comment by mallenexpensify** [comment]
### Payment Summary
Contributor+: @ishpaul777  due $1,500 via NewDot

@ishpaul777 , apologies for the delay, I'll be updating the BEC onboarding deets to state that any work started after being onboarded should be hourly, otherwise there's a grey area, which we don't want.

**Comment by JmillsExpensify** [comment]
$1,500 approved for @ishpaul777
